src/portable/ReactUi/react/src/components/UserSettings.tsx

```diff
@@ -418,36 +411,36 @@ class UserSettings extends React.Component<IProps, any> {
         const user = users.filter(u => u.username.id === selectedUser)[0];
 
         // Get the Default User HotKeys
-        this.defaultPlayPauseCode = userHotKeys.filter(h => h.id === "play-pause")[0].text;
-        this.defaultRewindCode = userHotKeys.filter(h => h.id === "back")[0].text;
-        this.defaultFastForwardCode = userHotKeys.filter(h => h.id === "forward")[0].text;
-        this.defaultSlowDownCode = userHotKeys.filter(h => h.id === "slower")[0].text;
-        this.defaultSpeedUpCode = userHotKeys.filter(h => h.id === "faster")[0].text;
+        this.defaultPlayPauseKey = userHotKeys.filter(h => h.id === "play-pause")[0].text;
+        this.defaultBackKey = userHotKeys.filter(h => h.id === "back")[0].text;
+        this.defaultForwardKey = userHotKeys.filter(h => h.id === "forward")[0].text;
+        this.defaultSlowerKey = userHotKeys.filter(h => h.id === "slower")[0].text;
+        this.defaultFasterKey = userHotKeys.filter(h => h.id === "faster")[0].text;
 
         // Set the Default User HotKeys in the corresponding Textboxes
-        if(context.playpauseRef.current !== null){
-            context.playpauseRef.current.state.message = this.keyCodeWithDefault(user, 'playpauseRef', this.defaultPlayPauseCode);
-            this.setState({inputValue: this.keyCodeWithDefault(user, 'playpauseRef', this.defaultPlayPauseCode)})
+        if(context.playPauseRef.current !== null){
+            context.playPauseRef.current.state.message = this.keyCode(user, 'playPauseRef', this.defaultPlayPauseKey);
```

Comment:
So the value is set to this.keycode the second parameter is the tag and is passed as 'playPauseRef'. On line 238 above, this parameter is compared to the id value in the user.hotkey array. The spec for the user hotkey array only has the values of 'play-pause', 'back', 'forward', 'slower', 'faster'. So I would expect that it will never find an element of user.hotkey array that has this tag and consequently will always user the default. But why not just set it to the default value if that is what we want? Why not just pass an empty string for the tag if we want the search to always fail. This same issue is true for the next line, and lines 427, 428, etc.

Reply:
Hi Greg. I have the changes that you have mentioned above and the have committed the code. The commit url is : https://github.com/manjula-t-ecgroup/Transcribe/commit/55d1a73af8ee77f02b739f5388c01154d7764aca